src/app/portlets/content-types/main/content-types.component.ts

```diff
@@ -1,119 +1,110 @@
+export class ContentTypesPortletComponent implements OnInit {
     public contentTypeColumns: DataTableColumn[];
     public item: any;
     public actionHeaderOptions: ActionHeaderOptions;
     public rowActions: MenuItem[];
+    public i18nMessages = {};
+    private i18nKeys = [
+        'contenttypes.fieldname.structure.name',
+        'contenttypes.content.variable',
+        'contenttypes.form.label.description',
+        'contenttypes.fieldname.entries',
+        'message.structure.delete.structure.and.content',
+        'message.structure.cantdelete',
+        'contenttypes.content.fileasset',
+        'contenttypes.content.content',
+        'contenttypes.content.persona',
+        'contenttypes.content.widget',
+        'contenttypes.content.htmlpage',
+        'contenttypes.content.key_value',
+        'contenttypes.content.vanity_url',
+        'contenttypes.content.form',
+        'contenttypes.confirm.message.delete',
+        'contenttypes.confirm.message.delete.content',
+        'contenttypes.confirm.message.delete.warning',
+        'contenttypes.action.delete',
+        'contenttypes.action.cancel',
+        'Content-Type'
+    ];
     @ViewChild('listing') listing: ListingDataTableComponent;
 
     constructor(
-        messageService: MessageService,
+        private messageService: MessageService,
         private router: Router,
         private route: ActivatedRoute,
         private contentTypesInfoService: ContentTypesInfoService,
         private crudService: CrudService,
-        private dotConfirmationService: DotConfirmationService
-    ) {
-        super(
-            [
-                'contenttypes.fieldname.structure.name',
-                'contenttypes.content.variable',
-                'contenttypes.form.label.description',
-                'contenttypes.fieldname.entries',
-                'message.structure.delete.structure.and.content',
-                'message.structure.cantdelete',
-                'contenttypes.content.file',
-                'contenttypes.content.content',
-                'contenttypes.content.persona',
-                'contenttypes.content.widget',
-                'contenttypes.content.page',
-                'contenttypes.confirm.message.delete',
-                'contenttypes.confirm.message.delete.content',
-                'contenttypes.confirm.message.delete.warning',
-                'contenttypes.action.delete',
-                'contenttypes.action.cancel',
-                'Content-Type'
-            ],
-            messageService
-        );
+        private dotConfirmationService: DotConfirmationService,
+        private dotContentletService: DotContentletService
+    ) {}
+
+    ngOnInit() {
+        Observable.forkJoin(
+            this.messageService.getMessages(this.i18nKeys),
+            this.dotContentletService.getAllContentTypes()
+        ).subscribe(val => {
+            this.i18nMessages = val[0];
```

Comment:
We can't save this here, I like the idea of not using the ```BaseComponent``` here, but save the messages in every component where we are going to use it it's not a good idea and it's not leaving in a better position of where we want to go... when you ask the the ```messageService``` for messages they are store in the ```messageService```, here: https://github.com/dotCMS/core-web/blob/master/src/app/api/services/messages-service.ts#L111

So what we need to do it's create a method in the message service like:

```
get(key: string): Observable<string> {
    return this.messagesLoaded.filter(SOMETHING HERE)
}
```

I think something like this could work... but we need to test.